File: backend/src/main/java/com/magesty/backend/models/Langue.java
package com.magesty.backend.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.magesty.backend.models.dto.AdminDto;
import com.magesty.backend.models.dto.LangueDto;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Objects;
import java.util.UUID;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Entity
public class Langue implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String level;

    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn( name = "langueId")
    private Admin admin;

    public static Langue from(LangueDto langueDto){
        Langue langue = new Langue();

        if(Objects.isNull(langueDto)){
            return null;
        }else{
            langue.setId(langueDto.getId());
            langue.setName(langueDto.getName());
            langue.setLevel(langueDto.getLevel());

            return langue;
        }
    }
}

File: frontend-ng/src/app/models/Langue/langue.ts
export class Langue {
  private _id: number;
  private _name: string;
  private _level: string;

  constructor(id: number, name: string, level: string) {
    this._id = id;
    this._name = name;
    this._level = level;
  }

  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get name(): string {
    return this._name;
  }

  set name(value: string) {
    this._name = value;
  }

  get level(): string {
    return this._level;
  }

  set level(value: string) {
    this._level = value;
  }
}
